core/Pos/Aeson/Core.hs

```diff
@@ -0,0 +1,62 @@
+-- | Aeson instances for core
+
+module Pos.Aeson.Core
+       (
+       ) where
+
+import           Universum
+
+import           Data.Aeson       (ToJSON (toJSON), FromJSON (..))
+import           Data.Aeson.TH    (defaultOptions, deriveToJSON, deriveJSON, deriveFromJSON)
+import           Data.Time.Units  (Millisecond, Microsecond)
+import           Serokell.Util.Base64  (JsonByteString (..))
+
+import           Pos.Aeson.Crypto ()
+import           Pos.Binary.Class (AsBinary (..))
+import           Pos.Core.Types   (ApplicationName (..), BlockCount (..), ChainDifficulty, CoinPortion,
+                                   Coin, EpochIndex, LocalSlotIndex, SharedSeed (..),
+                                   SlotCount (..), SlotId, BlockVersionData, SoftforkRule,
+                                   EpochIndex, Timestamp)
+import           Pos.Core.Fee     (TxFeePolicy, TxSizeLinear, Coeff)
+import           Pos.Core.Vss     (VssCertificate)
+
+instance FromJSON ByteString where
+    parseJSON v = getJsonByteString <$> parseJSON v
+
+instance ToJSON SharedSeed where
+    toJSON = toJSON . pretty
+
+instance FromJSON SharedSeed where
+    parseJSON v = SharedSeed <$> parseJSON v
+
+instance FromJSON (AsBinary w) where
+    parseJSON v = AsBinary <$> parseJSON v
+
+deriving instance ToJSON SlotCount
+
+-- NOTE: some of these types are used on frontend (PureScript).
+-- We are automatically deriving instances there and they are
+-- compitable now with `deriveToJSON defaultOptions ''Y`.
+-- If datatype is used on frontend, please use this instead of
+-- any other way of deriving if possible.
+
+deriveToJSON defaultOptions ''BlockCount
+deriveToJSON defaultOptions ''ApplicationName
+deriveToJSON defaultOptions ''ChainDifficulty
+deriveToJSON defaultOptions ''SlotId
+deriveToJSON defaultOptions ''LocalSlotIndex
+deriveToJSON defaultOptions ''EpochIndex
+
+deriveJSON defaultOptions ''Coin
+
+deriveFromJSON defaultOptions ''VssCertificate
+deriveFromJSON defaultOptions ''EpochIndex
+deriveFromJSON defaultOptions ''Millisecond
+deriveFromJSON defaultOptions ''Microsecond
+deriveFromJSON defaultOptions ''Timestamp
+deriveFromJSON defaultOptions ''CoinPortion
+deriveFromJSON defaultOptions ''SoftforkRule
+deriveFromJSON defaultOptions ''Coeff
+deriveFromJSON defaultOptions ''TxSizeLinear
+deriveFromJSON defaultOptions ''TxFeePolicy
+deriveFromJSON defaultOptions ''BlockVersionData
```

Comment:
Where did it come from actually? I see you added this stuff, but didn't remove it anywhere. It's suspicious.

Reply:
done.